packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb.tmpl

```diff
@@ -42,30 +42,12 @@ end
 #   target 'MyApp' do
 #     install_flutter_engine_pod
 #   end
-def install_flutter_engine_pod
-  current_directory = File.expand_path('..', __FILE__)
-  engine_dir = File.expand_path('engine', current_directory)
-  framework_name = 'Flutter.xcframework'
-  copied_engine = File.expand_path(framework_name, engine_dir)
```

Comment:
Stop copying the engine into the module ephemeral directory.  Let the tool handle copying the right version where it needs to go when `assemble` is called.